### DISCUSSION TITLE
Clarification about mNumGrantGlitch coex counter

### DISCUSSION THREAD

**Discussion by hemanth-silabs** [openthread/openthread#9693] [General]
@jwhui @abtink Can you please clarify the purpose of `mNumGrantGlitch` field in `otRadioCoexMetrics`? My understanding is that this indicates the number of times grant was asserted without request. Is my understanding correct?

**Comment by abtink** [comment]
@hemanth-silabs From my side I don't know enough about coex to be able to help you on this.

Generally the coex is delegated to the radio platform. OT stack itself is not involved it. The addition of `otRadioCoexMetrics` is only intended so that some helpful info (set of coex related counters) can retrieved from radio platform and passed through CLI and NCP.

**Reply by romacdon** [reply]
Thanks @abtink - @zhanglongxia it looks like you are the originator of `otRadioCoexMetrics`. Do you recall what you had in mind for the `mNumGrantGlitch` field?

**Reply by hemanth-silabs** [reply]
@jwhui @zhanglongxia any additional updates on this. The documentation around this field is not  very descriptive and we are wondering how to maintain this.

**Reply by zhanglongxia** [reply]
Hi [hemanth-silabs](https://github.com/hemanth-silabs). In general, when the GRANT pin enters the active state, we assume that the Thread radio should be able to transmit or receive. But sometimes, the GRANT pin will quickly leave the active state after entering the active state. We consider this case as the `GrantGlitch`. This case has the same effect as GRANT pin not active.

**Comment by Irving-cl** [comment]
Hi @hemanth-silabs,

`Glitch` referes to the cases where the grant pin was unintentionally set to a different value for a very short time. In these cases, the handler will be triggered but the handler will see that the grant pin value doesn't change so the handler won't take any action.

**Comment by hemanth-silabs** [comment]
Thanks for your response @zhanglongxia @Irving-cl. Can you elaborate on what is considered as quick/short time? Should we consider any abort happening after the GRANT is active as glitch as well? Should we define min grant time, below which abort is considered glitch?

**Reply by zhanglongxia** [reply]
We generally use a GPIO interrupt to monitor the GRANT pin. We will check the GRANT pin's state in the interrupt function, if GRANT pin's state doesn't match the interrupt edge settings, we think a `GrantGlitch` happened. The Coex metrics are implemented in the platform layer, the `min grant time` can be defined by the platform developers.

**Comment by hemanth-silabs** [comment]
Thanks for the inpurs @zhanglongxia @Irving-cl. I will close this discussion.
